mantidproject/mantid
# Preparation for Euphonic version bump: make compatible with both 0.6 and 1.*

**Description of work**
Make Abins and SimulatedDensityOfStates compatible with newer versions of Euphonic than currently pinned. At this point the pin is _not_ changed.

Clean up some of the legacy logic from this code.

**Purpose of work**
Mantid has pinned euphonic to 0.6.x, which is outdated. This has caused some dependency complications (#35797 ) and is blocking some Abins refactoring/features (e.g. #33755)

Rather than update the dependency, existing code and incorporate new features simultaneously, here we begin by making it possible to update the dependency without (hopefully) breaking anything that already exists.

**Summary of work**
- Prepare Mantid for API changes in Euphonic versions 1.x, ahead of updating dependency.
- Remove logic in SimulatedDensityOfStates and Abins that handle missing Euphonic library, as _some_ version of this dependency should always be available in current versions of Mantid. (This dates from the pre-Conda Mantid in which Euphonic was an optional dependency.)
- Remove complex temporary installation from Systemtest; again, this is no longer needed

**Further detail of work**
The API change isn't such a big problem to handle: one small function has been deprecated and as it wraps one that still exists we can just insert equivalent code.

**To test:**

With the `main` branch of Mantid, update Euphonic to a version > 1.0 using mamba. (The current release is 1.2.0). Run Mantid unit tests and systemtests; some of them should fail. (In particular the SimulatedDensityOfStates systemtests; to avoid running unrelated tests you could use `-R Abins` and `-R SimulatedDensity` arguments!)

On this branch, the tests should pass with both Euphonic 0.6.* and with Euphonic 1.*

*This does not require release notes* because **there are no user-facing changes, unless they are in the habit of installing incorrect package versions in their Mantid environment**

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
